src/result_processor.c

```diff
@@ -1457,7 +1457,8 @@ static int RPMaxScoreNormalizer_Accum(ResultProcessor *rp, SearchResult *r) {
  *******************************************************************************************************************/
 typedef struct {
   ResultProcessor base;             // Base result processor struct
-  RedisSearchCtx *sctx;             // Search context
+  RedisSearchCtx *depletingThread;  // Downstream Search context - used by the depleting thread
+  RedisSearchCtx *nextThread;       // Updstream search context - used by the thread calling Next
```

Comment:
You mixed the `down` and `up` :)
Also, let's add a word about the why these separate contexts are here.